### PR TITLE
Adding code coverage for unit tests

### PR DESCRIPTION
**Pull request by balda-rdx**

### CODE REVIEW

**Review by shambupujar — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -81,21 +81,21 @@ jobs:
         os: [k8s-linux-runner]
     steps:
     - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
+    - uses: radixdlt/public-iac-resuable-artifacts/rust-setup@feature/rust-coverage
```
Comment (lrubasze) [inline .github/workflows/ci.yml:84]: reusable?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -33,46 +32,47 @@ jobs:
...
     name: Run SBOR unit tests
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [k8s-linux-runner]
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: 1.70.0
```
Comment (lrubasze) [inline .github/workflows/ci.yml:46]: How is the rust version configured for code coverage tests?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (balda-rdx) [reply .github/workflows/ci.yml:84]: yeah there's a typo in the repo error. nothing we can do I'm afraid

File: .github/workflows/ci.yml (hunk 2 above)
Comment (balda-rdx) [reply .github/workflows/ci.yml:46]: good question. so if you take a look at the current pipeline you'll see that there are usually 2 steps
1. setup rust
2. run tests with coverage
In step 1 you can set the toolchain version, which currently defaults to 1.70.0: https://github.com/radixdlt/public-iac-resuable-artifacts/blob/main/rust-setup/action.yml#L4-L7

**Pull request merged**